Source/Core/Core/IOS/Network/IP/Top.cpp

```diff
@@ -155,48 +155,17 @@ IPCCommandResult NetIPTop::IOCtl(const IOCtlRequest& request)
     return GetDefaultReply(IPC_EACCES);
   }
 
-  s32 return_value = 0;
   switch (request.request)
   {
   case IOCTL_SO_STARTUP:
-  {
-    request.Log(GetDeviceName(), LogTypes::IOS_WC24);
-    break;
-  }
+    return StartUp(request);
   case IOCTL_SO_SOCKET:
-  {
-    u32 af = Memory::Read_U32(request.buffer_in);
-    u32 type = Memory::Read_U32(request.buffer_in + 4);
-    u32 prot = Memory::Read_U32(request.buffer_in + 8);
-
-    WiiSockMan& sm = WiiSockMan::GetInstance();
-    return_value = sm.NewSocket(af, type, prot);
-    INFO_LOG(IOS_NET, "IOCTL_SO_SOCKET "
-                      "Socket: %08x (%d,%d,%d), BufferIn: (%08x, %i), BufferOut: (%08x, %i)",
-             return_value, af, type, prot, request.buffer_in, request.buffer_in_size,
-             request.buffer_out, request.buffer_out_size);
-    break;
-  }
+    return Socket(request);
   case IOCTL_SO_ICMPSOCKET:
-  {
-    u32 pf = Memory::Read_U32(request.buffer_in);
-
-    WiiSockMan& sm = WiiSockMan::GetInstance();
-    return_value = sm.NewSocket(pf, SOCK_RAW, IPPROTO_ICMP);
-    INFO_LOG(IOS_NET, "IOCTL_SO_ICMPSOCKET(%x) %d", pf, return_value);
-    break;
-  }
+    return ICMPSocket(request);
   case IOCTL_SO_CLOSE:
   case IOCTL_SO_ICMPCLOSE:
-  {
-    u32 fd = Memory::Read_U32(request.buffer_in);
-    WiiSockMan& sm = WiiSockMan::GetInstance();
-    return_value = sm.DeleteSocket(fd);
-    INFO_LOG(IOS_NET, "%s(%x) %x",
-             request.request == IOCTL_SO_ICMPCLOSE ? "IOCTL_SO_ICMPCLOSE" : "IOCTL_SO_CLOSE", fd,
-             return_value);
-    break;
-  }
```

Comment:
GitHub won't let me directly put a comment on it, but I wasn't sure what to name the function for the code following this comment (for accept, bind, connect, and fcntl). If you have a naming suggestion, feel free to let me know and I'll move this code into a function with that name.